components/org.wso2.carbon.identity.handler.event.account.lock/src/main/java/org/wso2/carbon/identity/handler/event/account/lock/handlers/PostAuthnFailedLockoutClaimHandler.java

```diff
@@ -0,0 +1,168 @@
+
+    }
+
+    /**
+     * To get an instance of {@link PostAuthnFailedLockoutClaimHandler}.
+     *
+     * @return an instance of PostAuthnFailedLockoutClaimHandler.
+     */
+    public static PostAuthnFailedLockoutClaimHandler getInstance() {
+
+        return instance;
+    }
+
+    @Override
+    public int getPriority() {
+
+        int priority = super.getPriority();
+        if (priority == -1) {
+            priority = 21;
+        }
+        return priority;
+    }
+
+    @Override
+    public String getName() {
+
+        return handlerName;
+    }
+
+    @Override
+    public PostAuthnHandlerFlowStatus handle(HttpServletRequest request, HttpServletResponse response,
+                                             AuthenticationContext context) throws PostAuthenticationFailedException {
+
+        AuthenticatedUser authenticatedUser = context.getSequenceConfig().getAuthenticatedUser();
+        try {
+            if (authenticatedUser == null) {
+                return SUCCESS_COMPLETED;
+            }
+            /* Check whether account locking enabled and user is a local user.
+            Account locking is not done for federated flows. */
+            if (authenticatedUser.isFederatedUser() || isAccountLockingDisabled(authenticatedUser.getTenantDomain())) {
+                return SUCCESS_COMPLETED;
+            }
+            String usernameWithDomain = IdentityUtil.addDomainToName(
+                    authenticatedUser.getUserName(), authenticatedUser.getUserStoreDomain());
+            UserRealm realm = getUserRealm(authenticatedUser.getTenantDomain());
+            UserStoreManager userStoreManager = realm.getUserStoreManager();
+            Map<String, String> claimValues = getUserClaims(userStoreManager, usernameWithDomain);
+            String accountLockClaim = claimValues.get(ACCOUNT_LOCKED_CLAIM);
+            if (accountLockClaim != null && !Boolean.parseBoolean(accountLockClaim)) {
+                String failedLoginLockoutCount =
+                        claimValues.get(FAILED_LOGIN_LOCKOUT_COUNT_CLAIM);
+                // Check if the value is already not zero.
+                if (NumberUtils.isNumber(failedLoginLockoutCount) &&
+                        Integer.parseInt(failedLoginLockoutCount) > 0) {
+                    Map<String, String> updatedClaims = new HashMap<>();
+                    updatedClaims.put(FAILED_LOGIN_LOCKOUT_COUNT_CLAIM, "0");
+                    userStoreManager.setUserClaimValues(usernameWithDomain, updatedClaims, null);
+                }
+            }
+
+        } catch (UserStoreException e) {
+            // If user not found e, then continue.
+            if (!e.getMessage().contains(UserCoreErrorConstants.ErrorMessages.ERROR_CODE_NON_EXISTING_USER.getCode())) {
+                throw new PostAuthenticationFailedException(ERROR_WHILE_GETTING_USER_STORE_MANAGER_ERROR_CODE,
+                        "Error occurred while retrieving user store manager.");
+            }
```

Comment:
Check with the else clause

Reply:
The else clause will check for the user not existing error. That has been already handled in https://github.com/wso2/carbon-kernel/blob/2719cc17774dac1c5ad06aa34abfc6e4dc5e57f5/core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/common/AbstractUserStoreManager.java#L2125